lampod-cli/src/main.rs

```diff
@@ -91,68 +94,97 @@ async fn run(args: LampoCliArgs) -> error::Result<()> {
         _ => error::bail!("client {:?} not supported", client),
     };
 
-    if let Some(ref _private_key) = lampo_conf.private_key {
-        error::bail!("Option to force a private key not available at the moment")
-    }
-
-    let words_path = format!("{}/", lampo_conf.path());
-    // There are several case in this if-else, that are:
-    // 1. lampo is running on a fresh os:
-    //   1.1: the user has a wallet, so need to specify `--restore-wallet`
-    //   1.2: the user does not have a wallet so lampo should generate a new seed for it and store it in a file.
-    // 2. lampo is running on os where there is a wallet, lampo will load the seeds from wallet:
-    //   2.1: The user keep specify --restore-wallet, so lampo should return an error an tell the user that there is already a wallet
-    //   2.2: The user do not specify the --restore-wallet, so lampo load from disk the file, if there is no file it return an error
-    // FIXME: there is a problem of code duplication here, we should move this code in utils functions.
-    let wallet = if restore_wallet {
-        if Path::new(&format!("{}/wallet.dat", words_path)).exists() {
-            // Load the mnemonic from the file
-            let mnemonic = load_words_from_file(format!("{}/wallet.dat", words_path))?;
-            let wallet = match client.kind() {
+    let wallet = match (lampo_conf.private_key.clone(), lampo_conf.channels_keys.clone()) {
```

Comment:
Wow this match is really complex!

No you need to e explain me this sorry :) I am not able to simply understand while reading